### DISCUSSION TITLE
Interested in adding cloud storage for playlist and liked songs?

### DISCUSSION THREAD

**Discussion by shivardev** [sayaka-sh/spmp#306] [Ideas]
Hello, I will handle all the hosting and backend part. I need help with Kotlin as it is out of my expertise. If anyone is interested We can create email based backup system storing cloud. in form of JSON. I will handle cloud part as well. I need someone with Kotlin UI

**Comment by sayaka-sh** [comment]
I'm not interested in this myself, but you might find someone else if you ask on the discord server. 

Keep in mind that SpMp is published to F-Droid, so any dependencies will have to be FOSS for the feature to be merged (if that's what you want).

**Reply by shivardev** [reply]
So, FOSS won't allow cloud sync? I wanted to create a system where my playlist and liked songs sync between Rimusic and SpMp. would be a great feature. Guess I have to learn Kotlin. I request to give this feature some attention maybe some Kotlin dev will reach out to me!

**Reply by sayaka-sh** [reply]
Cloud sync itself is fine, you just can't use proprietary components (such as Firebase).

**Reply by shivardev** [reply]
I will create an open repo for backend Node / Go. will use free cloud services. superbase? or neon? firebase?.
This App will just make API calls. Firebase etc will not be installed in this App at all. 
This may I don't think this will violate FOSS roles for Fdroid
The app on changed activity on liked songs or something will create the JSON or some kind of TEXT-based list about the songs.. will send the playlist data to My backend through API. So that no Firebase SDK or anything will be included in the App repo.
